Feature comments

SinitsaBogdan opened the pull request.

SinitsaBogdan committed:
Update schema.sql

SinitsaBogdan committed:
Create Comment.java

SinitsaBogdan committed:
Create CommentDto.java

SinitsaBogdan committed:
Create RequestCommentDto.java

SinitsaBogdan committed:
Create CommentMapper.java

SinitsaBogdan committed:
Create CommentRepository.java

SinitsaBogdan committed:
Create CommentPublicController.java

SinitsaBogdan committed:
Create CommentPrivateController.java

SinitsaBogdan committed:
Create CommentAdminController.java

SinitsaBogdan committed:
Create CommentService.java

SinitsaBogdan committed:
Create CommentServiceImpl.java

SinitsaBogdan committed:
Create AccessDeniedException.java

SinitsaBogdan committed:
Update ExceptionsHandler.java

SinitsaBogdan committed:
Create feature.json

SinitsaBogdan committed:
Update README.md

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/comment/model/Comment.java:
```diff
@@ -0,0 +1,37 @@
+    private String text;
+
+    @Column(name = "timestamp")
+    private LocalDateTime createdOn;
```
Можно также добавить поле с датой изменения комментария, и менять его каждый раз когда будет меняться комментарий

SinitsaBogdan replied:
DONE

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/comment/controller/CommentAdminController.java:
```diff
@@ -0,0 +1,35 @@
+
+import javax.validation.Valid;
+
+@Slf4j
```
Не совсем понятно зачем тут данная аннотация, если логов в контроллере нет. Можно залогировать входящие запросы

SinitsaBogdan replied:
DONE

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/comment/controller/CommentAdminController.java:
```diff
@@ -0,0 +1,35 @@
+import javax.validation.Valid;
+
+@Slf4j
+@Validated
```
Аннотация validated нужна для проверки примитивов в самом контроллере, перед DTO уже стоит valid. Можно дополнительно поставить валидацию на некоторые параметры, чтобы сразу отсеивать невалидные значения

SinitsaBogdan replied:
DONE

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/comment/dto/RequestCommentDto.java:
```diff
@@ -0,0 +1,24 @@
+@AllArgsConstructor
+public class RequestCommentDto {
+
+    @NotNull
```
Нет особого смысла одновременно ставить две эти аннотации, так как NotBlank уже проверят на NotNull

SinitsaBogdan replied:
DONE

AlexeyCheuzov commented on ewm-main-service/src/main/java/ru/practicum/comment/repo/CommentRepository.java:
```diff
@@ -0,0 +1,13 @@
+
+import java.util.List;
+
+@Repository
```
Аннотация Repository не является обязательной, если Repository наследуется, к примеру, от JpaRepository. Можно её оставить, но лучше держать этот момент в голове, иногда спрашивают на собеседованиях.

SinitsaBogdan replied:
DONE

SinitsaBogdan committed:
Update repository

Аннотация Repository не является обязательной, если Repository наследуется, к примеру, от JpaRepository. Можно её оставить, но лучше держать этот момент в голове, иногда спрашивают на собеседованиях.

SinitsaBogdan committed:
Update RequestCommentDto.java

Нет особого смысла одновременно ставить две эти аннотации, так как NotBlank уже проверят на NotNull

SinitsaBogdan committed:
Update Valid Controller

Аннотация validated нужна для проверки примитивов в самом контроллере, перед DTO уже стоит valid. Можно дополнительно поставить валидацию на некоторые параметры, чтобы сразу отсеивать не валидные значения

SinitsaBogdan committed:
Update log in Controller

Не совсем понятно зачем тут данная аннотация, если логов в контроллере нет. Можно залогировать входящие запросы

SinitsaBogdan committed:
Create new param in model Comment

Можно также добавить поле с датой изменения комментария, и менять его каждый раз когда будет меняться комментарий
